core/engine/src/builtins/builder.rs

```diff
@@ -524,20 +524,25 @@ impl<'ctx> BuiltInBuilder<'ctx, OrdinaryObject> {
 }
 
 impl<'ctx> BuiltInBuilder<'ctx, Callable<Constructor>> {
+    /// Create a new builder for a constructor function setting the properties ahead of time for optimizations (less reallocations)
     pub(crate) fn from_standard_constructor<SC: BuiltInConstructor>(
```

Comment:
thought: is it possible to use const generics here if these values are known at compile time over a parameter?

```suggestion
    pub(crate) fn from_standard_constructor<SC: BuiltInConstructor, const M: usize, const SM: usize>(
```

Reply:
Hmmm... this may cause code bloat due to monomorphization.

Reply:
> Hmmm... this may cause code bloat due to monomorphization.

ooooh, that's true. They could be constants on the `BuiltInConstructor` trait instead, which may actually be nicer now that I think about it. The fields can still be left as constants, but that should remove any potential code bloat from monomorphization.

Reply:
Yeah I was wondering this, if with const genetics rust will need to internally create a new concrete type for each instance. I can change to const